src/providers/twitch/TwitchIrcServer.cpp

```diff
@@ -296,6 +302,31 @@ std::shared_ptr<Channel> TwitchIrcServer::getChannelOrEmptyByID(
     return Channel::getEmpty();
 }
 
+void TwitchIrcServer::bulkRefreshLiveStatus()
+{
+    QStringList userLogins;
+    this->forEachChannel([&userLogins](ChannelPtr chan) {
+        userLogins.push_back(chan->getName());
+    });
+
+    getHelix()->fetchStreams(
+        QStringList(), userLogins,
+        [this](std::vector<HelixStream> streams) {
+            for (const auto &stream : streams)
+            {
+                auto chan = this->getChannelOrEmpty(stream.userLogin);
+                if (chan->getType() == Channel::Type::None)
+                    continue;
+
+                auto twitchChan = dynamic_cast<TwitchChannel *>(chan.get());
+                twitchChan->parseLiveStatus(true, stream);
+            }
+        },
+        []() {
+            // failure
+        });
```

Comment:
While not really common, it's possible for the user to have more channels than the endpoint allows (100 per call). This should either return an explicit error about what happened or batch the requests like #3442 did [here](https://github.com/Chatterino/chatterino2/pull/3442/files#diff-7be3a1d599ba90aac80333967a0790247b66c02ec1980045039e96e798afd2c3R133). Maybe even move the helper to `src/util/Twitch`.

Reply:
Okay, ill move your `getChannelsInBatches` to `src/util/Twitch` and use it here as well. good catch.

Reply:
zneix informed me to not change this right now since I would have to make unrelated changes to other files and that this will be refactored later in a different PR. I just copy and pasted the batching technique into `TwitchIrcServer` right now to use it.